File: MBShopBE/Models/Product.cs

using System;
using System.Collections.Generic;

namespace MBShopBE.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Label { get; set; }
        public string Description { get; set; }
        public bool Sold { get; set; }
        public bool New { get; set; }
        public float Price { get; set; }
        public float PriceSold { get; set; }
        public int CategoryId { get; set; }
        public ICollection<Taille> Tailles { get; set; }
        public ICollection<Couleur> Couleurs { get; set; }
        public ICollection<Tag> Tags { get; set; }
        public string ImageURL { get; set; }
        public ICollection<ProdImage> ProdImages { get; set; }
        public DateTime Date { get; set; }

    }
}

File: MBShopFE/src/app/static/models/Product.ts
import { Tailles } from "./Tailles";
import { Couleurs } from "./Couleurs";

export class Product {
    id?: Number;
    label: String;
    description: String;
    sold: Boolean;
    new: Boolean;
    price: Number;
    priceSold: Number;
    categoryId: Number;
    tailles:Tailles[];
    couleurs:Couleurs[];
    tags:Tags[];
    imageURL: String;
    prodImages:ProdImages[];
    date: Date;

    constructor(id: Number, label: String, description: String, sold: Boolean, neww: boolean ,price: Number, priceSold: Number, categoryId: Number, imageURL: String){
        this.id=id;
        this.label=label;
        this.description=description;
        this.sold=sold;
        this.new=neww;
        this.price=price;
        this.priceSold=priceSold;
        this.categoryId=categoryId;
        this.imageURL=imageURL;
    }
}